pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -822,40 +823,25 @@ func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMoun
 
 	// only 1 of device list, device filter, device path filter and use all devices can be specified.  We prioritize in that order.
 	if len(osdProps.devices) > 0 {
-		deviceNames := make([]string, len(osdProps.devices))
-		for i, device := range osdProps.devices {
-			devSuffix := ""
-			if count, ok := device.Config[config.OSDsPerDeviceKey]; ok {
-				logger.Infof("%s osds requested on device %s (node %s)", count, device.Name, osdProps.crushHostname)
-				devSuffix += ":" + count
-			} else {
-				devSuffix += ":1"
-			}
-			if databaseSizeMB, ok := device.Config[config.DatabaseSizeMBKey]; ok {
-				logger.Infof("osd %s requested with DB size %sMB (node %s)", device.Name, databaseSizeMB, osdProps.crushHostname)
-				devSuffix += ":" + databaseSizeMB
-			} else {
-				devSuffix += ":"
-			}
-			if deviceClass, ok := device.Config[config.DeviceClassKey]; ok {
-				logger.Infof("osd %s requested with deviceClass %s (node %s)", device.Name, deviceClass, osdProps.crushHostname)
-				devSuffix += ":" + deviceClass
-			} else {
-				devSuffix += ":"
-			}
-			if md, ok := device.Config[config.MetadataDeviceKey]; ok {
-				logger.Infof("osd %s requested with metadataDevice %s (node %s)", device.Name, md, osdProps.crushHostname)
-				devSuffix += ":" + md
-			} else {
-				devSuffix += ":"
-			}
-			deviceID := device.Name
+		configuredDevices := []config.ConfiguredDevice{}
+		for _, device := range osdProps.devices {
+			id := device.Name
 			if device.FullPath != "" {
-				deviceID = device.FullPath
+				id = device.FullPath
 			}
-			deviceNames[i] = deviceID + devSuffix
+			cd := config.ConfiguredDevice{
+				ID:          id,
+				StoreConfig: config.ToStoreConfig(device.Config),
+			}
+			configuredDevices = append(configuredDevices, cd)
+		}
+		b, err := json.Marshal(configuredDevices)
```

Comment:
nit: I'd suggest readable vars, but maybe it's just my preference
```suggestion
		marshaledDevices, err := json.Marshal(configuredDevices)
```